src/extension.ts

```diff
@@ -121,12 +120,24 @@ async function getPRsPerSHAs(){
 				if (issuesBySHAs.length === 0) {
 					vscode.window.showErrorMessage("No search results. Try selecting a bigger piece of code or another file.");
 				} else {
-					issuesBySHAs.forEach((issue: { url: any; }) => {
+					issuesBySHAs.forEach(async (issue: { url: any; }) => {
 						const issueUrl = issue.url;
 	
-						octokit.request(`GET ${issueUrl}/comments`).then(octoresp => {
+						await octokit.request(`GET ${issueUrl}/comments`).then(async octoresp => {
 							// this paints the panel
-							watermelonPanel.currentPanel?.doRefactor({ command: "prs", data: octoresp.data})
+							octoresp.data.forEach(async (issue: { issue_url: any, title: string }) => {
+								const issueUrl = issue.issue_url;
+
+								await octokit.request(`GET ${issueUrl}`).then(octoresp2 => {
+									let prTitle ="";
+									prTitle = octoresp2.data.title;
+									issue.title = prTitle;
+								});
+
+								watermelonPanel.currentPanel?.doRefactor({ command: "prs", data: [issue]});
```

Comment:
Are we sending only one issue?
Should we stream them as they come?

Reply:
They stack up each other fine. The reason why this is a draft PR isn't because we're not showing all comments. It's because we're repeating PR Titles.

Reply:
Probably not the desired result

Reply:
Almost fixed, but we're getting a bunch of re-runs of `doRefactor` by design. Would need to solve this and we would have the desired result 100% as we expect it. 

If your'e curious and wanna help, take a look at the notes on `extension.ts` on lines 159, 160, and 168

Reply:
It is not "by design".
Do refactor is a bridge between the backend and frontend (node process and panel).
It was thought out to pass messages to the panel only.
We could have a streaming message service, kinda like a websocket.